### PR TITLE
Adding changes to windows cluster creation

### PR DESCRIPTION
**Pull request by sowmyav27**
Parameterizing cloud provider for windows so that we can create vxlan and Host GW with and without CP enabled.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_rke_cluster_provisioning.py [hunk 1]
```diff
@@ -1139,15 +1146,20 @@ def create_and_validate_custom_host(node_roles, random_cluster_name=False,
...
-                                  version=K8S_VERSION):
+                                  version=K8S_VERSION, 
+                                  cloudprovider=CLOUDPROVIDER_ENABLED):
     client = get_user_client()
     cluster_name = random_name() if random_cluster_name \
         else evaluate_clustername()
 
     if windows:
-        if windows_flannel_backend == "host-gw":
+        if windows_flannel_backend == "host-gw" and cloudprovider:
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:1156]: Isn't `cloudprovider` always be boolean `True` here ? `CLOUDPROVIDER_ENABLED` is a string and non-empty will always return True.

**Review by izaac — COMMENTED** [review]
A comment about the `CLOUDPROVIDER_ENABLED`

File: tests/validation/tests/v3_api/test_rke_cluster_provisioning.py (hunk 1 above)
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:1156]: i think you are right. Changing it and testing it set to false. I checked for "CLOUDPROVIDER_ENABLED" = True earlier.

Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:1156]: Works now with the latest changes. Tested it locally with cloudprovider = True

**Pull request closed without merging**